src/ip_notifier.cpp

```diff
@@ -130,7 +130,11 @@ struct ip_change_notifier_impl final : ip_change_notifier
 			[](SCNetworkReachabilityRef /*target*/, SCNetworkReachabilityFlags /*flags*/, void *info)
 			{
 				auto obj = static_cast<ip_change_notifier_impl*>(info);
-				obj->m_ios.post([obj]() { if (obj->m_cb) obj->m_cb(error_code()); });
+				if (!obj->m_cb) return;
```

Comment:
not just the clearing it, but also testing it. any access to ``obj`` probably should be done on the boost.asio ``io_service`` object, and hence in the main libtorrent thread.